lib/susedistribution.pm

```diff
@@ -17,11 +17,20 @@ use testapi qw(send_key %cmd assert_screen check_screen check_var get_var save_s
 
 
 sub handle_password_prompt {
-    if (!get_var("LIVETEST") && !get_var('LIVECD')) {
-        assert_screen "password-prompt";
+    my ($console) = @_;
+
+    return if get_var("LIVETEST") || get_var('LIVECD');
+    assert_screen "password-prompt";
+    if ($console eq 'hyperv-intermediary') {
+        type_string get_required_var('VIRSH_GUEST_PASSWORD');
+    }
+    elsif ($console eq 'svirt') {
+        type_string get_required_var check_var('VIRSH_VMM_FAMILY', 'hyperv') ? 'HYPERV_PASSWORD' : 'VIRSH_PASSWORD';
```

Comment:
I guess with this one we can get some issues, similar what we had for ternary operator on perl 5.18. Don't you mind to add some brackets to be on safe side for prod?

Reply:
Sure, I put brackets everywhere.